src/components/PrimitiveCheckbox/index.js

```diff
@@ -1,27 +1,27 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { uniqueId } from '../../../../libs/utils';
+import { uniqueId } from './../../libs/utils';
 import Label from './label';
 import StyledContainer from './styled/container';
 import StyledInnerContainer from './styled/innerContainer';
 
-export default class InputCheckbox extends Component {
+export default class PrimitiveCheckbox extends Component {
     constructor(props) {
         super(props);
         this.inputId = uniqueId('input-checkbox');
         this.inputRef = React.createRef();
     }
 
     componentDidMount() {
-        const { indeterminate } = this.props;
-        this.inputRef.current.indeterminate = indeterminate;
+        const { checked } = this.props;
+        this.inputRef.current.checked = checked;
     }
 
     componentDidUpdate(prevProps) {
-        const { indeterminate: prevIndeterminate } = prevProps;
-        const { indeterminate } = this.props;
-        if (prevIndeterminate !== indeterminate) {
-            this.inputRef.current.indeterminate = indeterminate;
+        const { checked: prevChecked } = prevProps;
+        const { checked } = this.props;
+        if (prevChecked !== checked) {
+            this.inputRef.current.checked = checked;
```

Comment:
We change how we use the component but we can't change the native approach to make this. I think we should maintain the indeterminate here.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#Indeterminate_state_checkboxes